Dockerfile

```diff
@@ -0,0 +1,28 @@
+## Go builder image
+FROM golang:1.20.3 as builder
+
+ENV GOPATH=/usr/local/go/bin
+
+WORKDIR /go
+
+COPY . .
+
+RUN CGO_ENABLED=0 go build -ldflags "-s" -a -installsuffix cgo .
+
+## Base image for production usage
+FROM alpine:3.14 as production
```

Comment:
This might be the somewhat biggest showstopper. `musl` gets less use overall than `libc`, and `ash` (the shell in alpine) is somewhat different than `bash`: https://www.geeksforgeeks.org/difference-between-ash-and-bash/

I would be tempted to use `bash`: https://hub.docker.com/_/bash or even `busybox` (although that'll get you `sh` rather than `bash`, so I'm hesitant). 

Without this project gathering metrics on its own and based purely on experience reports I've received over the years, as well as looking at the usage of `kubectl-slice` here in Github I would be tempted to say `bash`, the image, might be a better fit.

Another thing IMHO I would do here would be to have the image include `kubectl` even if its usage cannot be easily achieved with `kubeconfigs` that use a tool to authenticate. Same with `krew` (the plugin manager) but this IMHO has lesser priority, we could see how well the Docker image plays in the wild and do this paragraph after.

Reply:
What use case do you find that necessary to do?
IMHO I only needed this to run in CI - instead of running the go code out-of-the-box.  
So I would keep it clean, and not put kubeconfig + krew inside.